utils/duplicates.go

```diff
@@ -24,13 +26,27 @@ import (
 // CheckDuplicates returns error if the provided object slice contains multiple
 // objects sharing the same version/kind/namespace/name combination.
 func CheckDuplicates(objs []*unstructured.Unstructured) error {
-	seen := map[string]struct{}{}
+	seen := map[string]string{}
 	for _, o := range objs {
 		k := fmt.Sprintf("%s, %q, %q", o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName())
-		if _, found := seen[k]; found {
+		v := hash(o)
+		if h, found := seen[k]; found {
+			// allow but elide literal duplicates
+			if h == v {
+				continue
+			}
 			return fmt.Errorf("duplicate resource %s", k)
 		}
-		seen[k] = struct{}{}
+		seen[k] = v
 	}
 	return nil
 }
+
+func hash(obj *unstructured.Unstructured) string {
+	h := sha1.New()
+	// ignore error based on the unvalidated assumption that we already have a valid, marshallable unstructured object
```

Comment:
Is this assumption valid? I'd prefer not to have to deal with the pointer dereferencing required in order to make this function return `(*string, error)`.

Reply:
Yes we have marshallable unstructured objects

Reply:
It's ok to not return the error but at least we could panic if we do receive an error (for some reason we cannot foresee here)

Reply:
Sure.

Reply:
PTAL